### PR TITLE
Tests for enqueue barrier extension

### PR DESCRIPTION
**Pull request by vladimirkhashev**
Added tests according the to test plan [enqueue_barrier.asciidoc](https://github.com/KhronosGroup/SYCL-CTS/blob/SYCL-2020/test_plans/enqueue_barrier.asciidoc).

### CODE REVIEW

File: tests/extension/oneapi_enqueue_barrier/enqueue_barrier.cpp [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+      q1.submit([&](sycl::handler& cgh) {
+          cgh.single_task([=] {
+            completion_flags[0] = false;
+            completion_flags[1] = false;
+          });
+        }).wait();
+
+      auto event1 = q1.submit([&](sycl::handler& cgh) {
+        auto result_buf_a_acc =
+            result_buf_a.get_access<sycl::access_mode::write>(cgh);
```
Comment (keryell) [inline tests/extension/oneapi_enqueue_barrier/enqueue_barrier.cpp:88]: Use `sycl::accessor` constructors everywhere instead, as it is the preferred SYCL 2020 way.

**Review by keryell — CHANGES_REQUESTED** [review]

File: tests/extension/oneapi_enqueue_barrier/enqueue_barrier.cpp (hunk 1 above)
Comment (vladimirkhashev) [reply tests/extension/oneapi_enqueue_barrier/enqueue_barrier.cpp]: Done.

File: tests/extension/oneapi_enqueue_barrier/enqueue_barrier.cpp [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+            completion_flags[0] = false;
+            completion_flags[1] = false;
+          });
+        }).wait();
+
+      auto event1 = q1.submit([&](sycl::handler& cgh) {
+        auto result_buf_a_acc =
+            sycl::accessor(result_buf_a, cgh, sycl::write_only);
+        auto loop_arr_buf_a_acc =
+            sycl::accessor(loop_arr_buf_a, cgh, sycl::write_only);
```
Comment (keryell) [inline tests/extension/oneapi_enqueue_barrier/enqueue_barrier.cpp:90]: While quite legal, why not simplifying everywhere with
```suggestion
        sycl::accessor result_buf_a_acc(result_buf_a, cgh, sycl::write_only);
        sycl::accessor loop_arr_buf_a_acc(loop_arr_buf_a, cgh, sycl::write_only);
```
?

File: tests/extension/oneapi_enqueue_barrier/enqueue_barrier.cpp [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+      q1.submit([&](sycl::handler& cgh) {
+          cgh.single_task([=] {
+            completion_flags[0] = false;
+            completion_flags[1] = false;
+          });
+        }).wait();
+
+      auto event1 = q1.submit([&](sycl::handler& cgh) {
+        sycl::accessor result_buf_a_acc(result_buf_a, cgh, sycl::write_only);
+        sycl::accessor loop_arr_buf_a_acc(loop_arr_buf_a, cgh, sycl::write_only);
```
Comment (bader) [inline tests/extension/oneapi_enqueue_barrier/enqueue_barrier.cpp:88]:
```suggestion
        sycl::accessor result_buf_a_acc(result_buf_a, cgh, sycl::write_only);
        sycl::accessor loop_arr_buf_a_acc(loop_arr_buf_a, cgh,
                                          sycl::write_only);
```

**Review by bader — APPROVED** [review]

**Comment by bader** [comment]
@keryell, I addressed your comments and fixed formatting. Let me know if anything else is needed here.

**Review by keryell — APPROVED** [review]
Thanks.

**Pull request merged**